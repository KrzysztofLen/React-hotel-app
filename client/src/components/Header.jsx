import React from 'react';
import LoginSystem from './LoginSystem/LoginSystem';
import {Logo} from './Logo/Logo';
import {HotelsFilterContainer} from "./Search/Search";
import {getFilteredHotels} from "../selectors/getFilteredHotels";
import {connect} from "react-redux";

export class Header extends React.Component {

	render() {
		return (
			<header className="header">
				<Logo/>
				{/*<Search/>*/}
				<HotelsFilterContainer/>
				<LoginSystem/>
			</header>
		)
	}
}

const mapStateToProps = state => {
	return {
		hotels: getFilteredHotels(state.hotels, state.hotelsSearch)
	};
};

export default connect(mapStateToProps, null)(Header);
