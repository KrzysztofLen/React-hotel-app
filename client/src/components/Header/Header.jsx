import React, {Component} from 'react';
import LoginSystem from '../LoginSystem/LoginSystem';
import {Logo} from '../Logo/Logo';
import {getFilteredHotels} from "../../selectors/getFilteredHotels";
import {connect} from "react-redux";
import Search from "../Search/Search";
import ModalWindow from "../ModalWindow";
import {Check} from "../SVG/Check";
import Success from "../Success/Success";

export class Header extends Component {
	constructor(props) {
		super(props);

		this.onModalOpen = this.onModalOpen.bind(this);
		this.onCloseModal = this.onCloseModal.bind(this);

		this.state = {
			modalIsOpen: false
		};
	}

	onModalOpen() {
		this.setState({modalIsOpen: true});
	}

	onCloseModal() {
		this.setState({modalIsOpen: false});
	}

	render() {
		return (
			<header className="header">
				<Logo/>
				<Search/>
				<button className="button is-success" onClick={this.onModalOpen}>Login<Check width={20} height={20} /></button>
				<ModalWindow isOpen={this.state.modalIsOpen} closeModal={this.onCloseModal} component={<LoginSystem/>}/>
			</header>
		)
	}
}

const mapStateToProps = (state: Object) => {
	return {
		hotels: getFilteredHotels(state.hotels, state.hotelsSearch)
	};
};

export default connect(mapStateToProps, null)(Header);
