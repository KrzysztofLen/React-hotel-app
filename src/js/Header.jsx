import React from 'react';
import LoginButton from "./LoginSystem/LoginButton";
import LogoutButton from './LoginSystem/LogoutButton';
import {Logo} from './Logo';

export default class Header extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			isLogged: false
		}
	}

	// console.log(startLogin);
	handleClick = () => {
		console.log('click');
		// this.setState({isLogged: true})
	};

	render() {
		return (
			<header className="header">
				<Logo/>
				<form action="#" className="search">
					<input type="text" className="search__input" placeholder="Search hotels"/>
				</form>
				<LogoutButton/>
				<LoginButton />
			</header>
		)
	}
};

