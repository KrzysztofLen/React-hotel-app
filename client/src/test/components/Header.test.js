import React from 'react';
import {shallow} from 'enzyme';
import {Header} from '../../components/Header++DELETE/Header';

test('should render Header++DELETE correctly', () => {
	const wrapper = shallow(<Header/>);
	expect(wrapper).toMatchSnapshot();
});