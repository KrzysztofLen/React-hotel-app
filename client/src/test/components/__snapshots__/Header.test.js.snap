// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <Logo />
  <form
    action="#"
    className="search"
  >
    <input
      className="search__input"
      placeholder="Search hotels"
      type="text"
    />
  </form>
  <LogoutButton />
  <LoginButton />
</header>
`;
